### DISCUSSION TITLE
Clarification on new Transmission while another one is in progress

### DISCUSSION THREAD

**Discussion by testter2** [openthread/openthread#9502] [Q&A] [answered]
Hi All,

I am currently working with OpenThread and have a question regarding concurrent transmissions. Specifically, I'm interested in understanding how OpenThread handles situations where there is a need to initiate a new transmission while another transmission is already in progress.

I am working on a radio platform that supports CSMA. It seems that before the CSMA backoff delay fires, another frame is sent, and the radio platform returns an "invalid_state" error because the device is not in the receive state. I've noticed that there's a line of code `SuccessOrAssert(error);` at end of `SubMac::BeginTransmit(void)` that checks for an error return value that's not "success."

Is there any possible way to start a transmission while there's another one in progress?

Thanks in advance.

**Comment by jwhui** [comment, marked as answer]
OpenThread only supports one transmit operation at a time and should not call `otPlatRadioTransmit()` unless the radio is in the receive state.

**Reply by testter2** [reply]
Thanks for quick reply.
